src/platform/backends/qemu/qemu_virtual_machine.cpp

```diff
@@ -343,31 +344,57 @@ void mp::QemuVirtualMachine::start()
     vm_process->write(qmp_execute_json("qmp_capabilities"));
 }
 
-void mp::QemuVirtualMachine::shutdown()
+void mp::QemuVirtualMachine::shutdown(const bool force)
 {
-    if (state == State::suspended)
+    std::unique_lock<std::mutex> lock{state_mutex};
+
+    try
     {
-        mpl::log(mpl::Level::info, vm_name, fmt::format("Ignoring shutdown issued while suspended"));
+        check_state_for_shutdown(force);
     }
-    else
+    catch (const VMStateIdempotentException& e)
     {
-        drop_ssh_session();
+        mpl::log(mpl::Level::info, vm_name, e.what());
+        return;
+    }
+
+    if (force)
+    {
+        mpl::log(mpl::Level::info, vm_name, "Forcing shutdown");
 
-        if ((state == State::running || state == State::delayed_shutdown || state == State::unknown) && vm_process &&
-            vm_process->running())
+        if (vm_process)
         {
-            vm_process->write(qmp_execute_json("system_powerdown"));
-            vm_process->wait_for_finished(timeout);
+            mpl::log(mpl::Level::info, vm_name, "Killing process");
+            lock.unlock();
+            vm_process->kill();
+            if (vm_process != nullptr && !vm_process->wait_for_finished(timeout))
```

Comment:
I don't think the same timeout should apply here as in normal shutdown/suspend situations. Note that this is intentionally high (5 min) to give QEMU a long time to quit properly and avoid corrupting the image. But here the intention is to exit immediately regardless of image damage. 

`QProcess::kill()` (which is what the above ends up calling) tries to send a SIGKILL. If the process does not quit immediately, it is unlikely to quit anytime. I don't think there is any point in waiting for any longer than 5s, just in case the machine is very busy/slow.

Reply:
Indeed, 5s is a more reasonable timeout value for this particular case.